sdk/storage/azure-storage-files-shares/test/ut/share_directory_client_test.cpp

```diff
@@ -834,4 +834,157 @@ namespace Azure { namespace Storage { namespace Test {
     {
     }
   }
-}}} // namespace Azure::Storage::Test
+
+  TEST_F(FileShareDirectoryClientTest, AllowTrailingDot)
+  {
+    const std::string directoryName = RandomString();
+    const std::string directoryNameWithTrailingDot = directoryName + ".";
+    const std::string connectionString = StandardStorageConnectionString();
+    const std::string shareName = m_shareName;
+    auto options = InitStorageClientOptions<Files::Shares::ShareClientOptions>();
+
+    auto testTrailingDot = [&](bool allowTrailingDot) {
```

Comment:
Besides explicit value (`true` and `false`), we also want to test the default behavior (not set).
This is very important to make sure there's not breaking change.

Reply:
I think existing unit tests such as `FileShareFileClientTest::CreateDeleteFiles`, `FileShareFileClientTest::DownloadEmptyFile` will not set value when running the latest code. Is it necessary to add  a unit test to test this case(not set value)?

Reply:
File name in those test cases doesn't end with dot (`.`), does it?

Reply:
Yeah, I will add this test case.